#!/bin/sh /usr/share/dpatch/dpatch-run
## 20_safe_path_at_init.dpatch by Henrique de Moraes Holschuh <hmh@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Set a safe path at startup for benefit of the config scripts

@DPATCH@
diff -urNad amavisd-new~/amavisd amavisd-new/amavisd
--- amavisd-new~/amavisd	2007-06-27 12:43:00.000000000 +0200
+++ amavisd-new/amavisd	2007-09-17 22:23:08.000000000 +0200
@@ -10939,6 +10939,10 @@
 my($desired_user);                       # username or UID
 if ($> != 0) { $desired_user = $> }      # use effective UID if not root
 
+# Use a default, guaranteed safe path during startup, before loading
+# an user-supplied one from the config file
+$ENV{PATH} = "/bin:/usr/bin";
+
 # collect and parse command line options
 my($log_level_override, $max_servers_override);
 my($myhome_override, $tempbase_override, $helpers_home_override);
